src/curve.ts

```diff
@@ -27,7 +43,7 @@ export class Curve extends Element {
   //    cps: List of control points
   //    x_shift: pixels to shift
   //    y_shift: pixels to shift
-  constructor(from, to, options) {
+  constructor(from: StaveNote, to: StaveNote, options: StaveOptions) {
```

Comment:
`StaveOptions` here seems wrong, particularly since we're extending `element`.

Also `from:` and `to:` should be `Note`, you can have curves on other elements (`GraceNoteGroup`, `TabNote`, etc.)

Reply:
A few attributes from `options` are used: `x_shift`, `y_shift`, thickness`, ... but not all. It would make sense to **create a `CurveOptions`** . It is used a parameter and as attribute from `Curve`.

`to` and `from` as `Note` would not have the following methods used by `Curve`:
From `StaveNote`: `getTieRightX` & `getTieLeftX`
From `StemmableNote`: `getStemDirection` & `getStemExtents`
Due to the fact that we do not want to merge `Note`, `StemmableNote` & `StaveNote`, it would make sense to **define those methods in `Note` returning an appropriated value**.

@0xfe do you agree?

Reply:
Okay, let's create a new `CurveOptions` with just what we need.

Re: `Note`, good catch, we should do two things:

- Move `getTieRightX()` and `getTieLeftX()` out of `TabNote` and into `Note` (the default implementation there will work everywhere.)
- Add a new curve option `direction?: enum = { -1: DOWN, 0: DEFAULT, 1: UP }` default to `0`. If the note is not a `StemmableNote` then use the `direction`.

Reply:
Done `getStemDirection` `getStemExtents` also implemented in `Note` throwing an error.